src/actions/validators.js

```diff
@@ -9,16 +10,20 @@ const ValidatorsActions = {
   findAll() {
     return async function(dispatch) {
       const jurisdiction = await Jurisdiction.at(JURISDICTION_ADDRESS)
+      // FIXME this is very slow. We have to add a function to the Jurisdiction
+      // contract that returns all the validators.
+      // See https://github.com/TPL-protocol/tpl-contracts/issues/2
+      // --elopio - 20180704
       const events = jurisdiction.ValidatorAdded({}, { fromBlock: 0, toBlock: 'latest' });
       events.watch(function(error, result) {
-        if(error) throw error
+        if(error) AlertActions.showError(error)
         else dispatch(ValidatorsActions.add(result.args.validator))
       })
     }
   },
 
   add(validator) {
-    return async function(dispatch) {
+    return function(dispatch) {
```

Comment:
Removing the `async` was not what I meant.

The *action* is still asychronous because it returns a function. This is so because of the Redux Thunk middleware we're using.

To be synchronous, `ValidatorsActions.add` has to return the action object itself.

Reply:
okay, I need to read in more detail the async section of the redux docs.